packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,409 @@
+    function withdraw() external {
+        // get caller's balance
+        uint256 balance = balances[msg.sender];
+
+        // set caller's balance to 0
+        balances[msg.sender] = 0;
+
+        // send caller's balance to caller
+        bool success = SafeCall.send(msg.sender, gasleft(), balance);
+        if (!success) {
+            revert WithdrawalFailed();
+        }
+    }
+
+    /// @notice Checks if the current block is within the challenge window for a given challenged block number.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @return True if the current block is within the challenge window, false otherwise.
+    function _isInChallengeWindow(uint256 challengedBlockNumber) internal view returns (bool) {
+        return (block.number >= challengedBlockNumber && block.number <= challengedBlockNumber + challengeWindow);
+    }
+
+    /// @notice Checks if the current block is within the resolve window for a given challenge start block number.
+    /// @param challengeStartBlockNumber The block number at which the challenge was initiated.
+    /// @return True if the current block is within the resolve window, false otherwise.
+    function _isInResolveWindow(uint256 challengeStartBlockNumber) internal view returns (bool) {
+        return block.number <= challengeStartBlockNumber + resolveWindow;
+    }
+
+    /// @notice Returns the status of a challenge for a given challenged block number and challenged commitment.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param challengedCommitment The commitment that is being challenged.
+    /// @return The status of the challenge.
+    function getChallengeStatus(
+        uint256 challengedBlockNumber,
+        bytes calldata challengedCommitment
+    )
+        public
+        view
+        returns (ChallengeStatus)
+    {
+        Challenge memory _challenge = challenges[challengedBlockNumber][challengedCommitment];
+        // if the address is 0, the challenge is uninitialized
+        if (_challenge.challenger == address(0)) return ChallengeStatus.Uninitialized;
+
+        // if the challenge has a resolved block, it is resolved
+        if (_challenge.resolvedBlock != 0) return ChallengeStatus.Resolved;
+
+        // if the challenge's start block is in the resolve window, it is active
+        if (_isInResolveWindow(_challenge.startBlock)) return ChallengeStatus.Active;
+
+        // if the challenge's start block is not in the resolve window, it is expired
+        return ChallengeStatus.Expired;
+    }
+
+    /// @notice Require the type of a commitment to be known.
+    /// @dev The type of a commitment is stored in its first byte.
+    /// @param commitment The commitment for which to check the type.
+    /// @return The commitment type of the given commitment.
+    function _requireKnownCommitmentType(bytes calldata commitment) internal pure returns (CommitmentType) {
```

Comment:
I am going back and forth a lot of this function, my first intuition is that it should accept a `uint8` but i suppose this is a simple way to do things so not super opinionated, just thinking out loud

Reply:
This should also check the size of the commitment?

Reply:
could change this to be `_validateCommitmentType` and for a known commitment type and for the max commitment size

Reply:
initial reason for passing in the entire commitment was to avoid having to do `(uint8(bytes1(commitment))` everywhere it's used, but agree on also checking the length of the commitment, which would be another reason to pass in the entire commitment

Reply:
updated